llvm/lib/Target/AArch64/AArch64ISelDAGToDAG.cpp

```diff
@@ -1130,10 +1139,6 @@ bool AArch64DAGToDAGISel::SelectAddrModeUnscaled(SDValue N, unsigned Size,
     return false;
   if (ConstantSDNode *RHS = dyn_cast<ConstantSDNode>(N.getOperand(1))) {
     int64_t RHSC = RHS->getSExtValue();
-    // If the offset is valid as a scaled immediate, don't match here.
```

Comment:
And this doesn't alter any of the existing codegen, for non-atomic types?

Reply:
Does not seem to be the case from running current codegen tests, at the very least.